src/main/java/spoon/support/reflect/declaration/CtElementImpl.java

```diff
@@ -158,23 +161,39 @@ public CtElementImpl() {
 		super();
 	}
 
+	/** the ordering  between CtElement is lexicographic,
+	 * based on the deep representation
+	 * which is also used in {@link #equals(Object)}.
+	 */
 	public int compareTo(CtElement o) {
-		String current = getSignature();
-		String other = o.getSignature();
+		String current = getDeepRepresentation(this);
+		String other = getDeepRepresentation(o);
 		if (current.length() <= 0 || other.length() <= 0) {
 			throw new ClassCastException("Unable to compare elements");
 		}
 		return current.compareTo(other);
 	}
 
+	public String getDeepRepresentation(CtElement elem) {
```

Comment:
This doesn't work. e.g., when you'll build the deep representation of a method like this one:

```java
public void toProvider(Provider<? extends T> provider) {
}
```

The result of `getDeepRepresentation` will be: `void toProvider(javax.inject.Provider,)voidprovider{}`.